### ISSUE TITLE
Report overwritten if multiple files

### ISSUE THREAD

**Issue by jheadley** [aj-dev/jscs-html-reporter#1]
If you have jscs scanning multiple files and more than one file returns results, the html report is overwritten each time so that the final report only contains details from the last file found to have issues. It would also be helpful if the target filename could be specified so that the location and name of the html report could be configured.

**Comment by aj-dev** [comment]
@jheadley can you clarify a few things:
1. Do you run multiple instances of JSCS commands at the same time or just a single instance?
2. Are you using reporter with plain node JSCS or as a gulp/grunt plugin?

I could not reproduce it when running a single instance of node JSCS. Make sure that `max-errors` is set to `null` or `-1` - http://jscs.info/overview#-max-errors-m. Could it be that you have more errors in your file than defined by `max-errors`? In that case, JSCS will report errors from that one file only, disregarding any other files with errors.

As for adding support for report target filename, I can look into that.

**Comment by jheadley** [comment]
Thanks for responding. I added the max errors to our .jscsrc file but still see the same results with just one file mentioned in the report.

Using gulp. I get output like this:
[14:50:08] Using gulpfile /git-repo/chem/gulpfile.js
[14:50:08] Starting 'jscs'...
>> jscs report written to /git-repo/chem/jscs-html-report.html
>> jscs report written to /git-repo/chem/jscs-html-report.html
>> jscs report written to /git-repo/chem/jscs-html-report.html

Our configuration is like this:
gulp.task('jscs', function() {
    return gulp.src(config.app + 'scripts/app/**/*.js')
        .pipe(jscs())
        .pipe(jscs.reporter('jscs-html-reporter')) ;
});

**Comment by aj-dev** [comment]
This is indeed a bug when using gulp. I completely forgot that `gulp-jscs` works with streams and it's very different from `node-jscs` or `grunt-jscs`. I'll see what I can do to fix it.

**Comment by aj-dev** [comment]
@jheadley I have released a beta version with a workaround for Gulp. Due to the implementation details of `gulp-jscs`, it is impossible to create one report for multiple files, so I exposed a standalone Gulp reporter which you can use in the following way.

```
var gulp = require('gulp');
var jscs = require('gulp-jscs');
var jscsGulpHtmlReporter = require('jscs-html-reporter').gulpReporter;

gulp.task('default', function () {
    return gulp.src('src/app.js')
               .pipe(jscs())
               .pipe(jscsGulpHtmlReporter({reporterOutput: '/path/to/jscs-html-report.html'}))
});
```

Optional output path can be provided with `{reporterOutput: '/path/to/jscs-html-report.html'}`.

You can install beta version with `npm i jscs-html-reporter@beta --save-dev`.

Let me know if that works for you.

**Comment by jheadley** [comment]
That seems to be working. Thanks!

**Comment by aj-dev** [comment]
Fixed in [v0.2.0](https://github.com/aj-dev/jscs-html-reporter/tree/v0.2.0) release.
